### DISCUSSION TITLE
How about autodetection_timeout

### DISCUSSION THREAD

**Discussion by DarkDau** [netdata/netdata#15278] [Q&A] [answered]
From time to time certain charts disappear in netdata. 
There are several reasons. Maybe there was no data from the monitored service for a while. In my case the complete chart for Apache was down for 2 days.
Normally one expects that after a certain time a new attempt is made.
But this is not the case generally.
I have now set global **autodetection_timeout = 120** in /etc/netdata/netdata.conf in general. Is this the right way?

**Comment by ilyam8** [comment, marked as answer]
Hi, @DarkDau. There is no such option as `autodetection_timeout`.

How it works:

- On start: 
    - Netdata tries to collect data from configured Apache instances.
    - If an instance is not available/doesn't return expected data Netdata doesn't retry until the next restart.
    - If an instance is available and returns data Netdata will continue to collect metrics.
- Runtime: 
    - If an instance was available but stopped responding at some point, Netdata will retry indefinitely and charts will appear as soon as the instance starts responding.

> On start: If an instance is not available/doesn't return expected data Netdata doesn't retry until the next restart.

Two exceptions:
 - If the instance was available before Netdata was restarted, Netdata will not issue the first failed attempt but will attempt 10 attempts at 30-second intervals.
 - This behavior can be changed by setting `autodetection_retry` to a non-zero value in the collector's configuration file (go.d/apache.conf in your case). See [autodetection_retry](https://github.com/netdata/go.d.plugin/blob/b2b7637c59ad51f98eaec93285890fb6d7b76900/config/go.d/apache.conf#L23-L25).

**Reply by DarkDau** [reply]
so, sadly autodetection_timeout is not an option. (because it does not exist)
Ok then I add autodetection_retry.
Thanx  for clearing this
